### PR TITLE
[matchit] Add first draft port

### PR DESCRIPTION
**Pull request by daljit97**
Add matchit, a library for pattern matching for C++17

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
I am still working on this PR

### CODE REVIEW

**Comment by BowenFu** [comment]
@daljit97 maybe the tests failed because you did not run ./vcpkg x-add-version --all?

**Comment by daljit97** [comment]
@BowenFu you're right I forgot to that, but that doesn't seem the issue. The problem seems to be that the files matchitConfig.cmake and matchitConfigVersion.cmake aren't generated in the buildtree of vcpkg (they are when running cmake ordinarily).

**Comment by BowenFu** [comment]
@daljit97 maybe the problem is that make install was not supported in v1.0.0, will add new version v1.0.1 to fix this.

**Comment by Cheney-W** [comment]
If this is a header-only library, you can refer to [this file](https://github.com/microsoft/vcpkg/blob/master/ports/websocketpp/portfile.cmake) to modify your PR.

**Comment by daljit97** [comment]
@BowenFu thanks indeed that seemed to be the issue (I probably should've checked against the 1.0.0 tag instead of head). I've tested on Linux and Windows and the port seems to work (at least locally).

File: ports/matchit/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,23 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO BowenFu/matchit.cpp
+    REF v1.0.1
+    SHA512 60edc6a392f5629391fa9e3ff09b7b98a0a782919a066ad2999eabb58e60f38bd50e080037b1276c5bca986f81ca0dfff2914816d46458b7b4e1c947a6134169
+    HEAD_REF master
```
Comment (BowenFu) [inline ports/matchit/portfile.cmake:6]:
```suggestion
    HEAD_REF main
```

File: ports/matchit/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+vcpkg_cmake_install()
+
+vcpkg_cmake_config_fixup(CONFIG_PATH "lib/cmake/matchit")
+
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/lib")
+
+file(INSTALL "${CMAKE_CURRENT_LIST_DIR}/usage" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")
+file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
```
Comment (Cheney-W) [inline ports/matchit/portfile.cmake:23]:
```suggestion
file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
```

Comment (daljit97) [reply ports/matchit/portfile.cmake]: @Cheney-W Committed.

**Comment by JavierMatosD** [comment]
Thank you!

**Pull request merged**